### DISCUSSION TITLE
How to manually add a browser?

### DISCUSSION THREAD

**Discussion by Taantawan** [U-C-S/Hurl#81] [General]
I added the lines below to "Browsers" of UserSettings.json and expected Vivaldi was going to show up on the Hurl popup window, but it does not.

What am I missing?

```
    {
      "Name": "Vivaldi",
      "ExePath": "C:\\Users\\tanaka\\AppData\\Local\\Vivaldi\\Application\\vivaldi.exe"
    },
``` 

On Windows 11

**Comment by U-C-S** [comment]
Have you "reload"/restarted the app ? 
For reload, Right-click on Hurl icon in Tray menu and use the reload option

I'm working on Hurl Settings app to avoid doing this in the future

**Reply by Taantawan** [reply]
No, I haven't. That is the culprit. The reload resolved my stupid issue. Thank you. 😊

**Reply by skyhirider** [reply]
@U-C-S Windows has an api where you can subscribe to file changes (if the files are on a local disk). You could subscribe to the config file and after its changed, ask if the app should reload itself.

**Reply by U-C-S** [reply]
@skyhirider I need something like that, Thanks

**Reply by skyhirider** [reply]
@U-C-S I've used this in Java on a Windows machine, but your tool is not written in that language
- https://docs.oracle.com/javase/tutorial/essential/io/notification.html
- https://www.baeldung.com/java-nio2-watchservice

Found generic documentation for C++, you could have a look at that
https://learn.microsoft.com/en-us/windows/win32/fileio/obtaining-directory-change-notifications

In general, you subscribe to specific events and get a callback once its triggered (copy paste from java docs)
ENTRY_CREATE – A directory entry is created.
ENTRY_DELETE – A directory entry is deleted.
ENTRY_MODIFY – A directory entry is modified.
OVERFLOW – Indicates that events might have been lost or discarded. You do not have to register for the OVERFLOW event to receive it.
